src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let men = 0;
+
+  return people.reduce((sum, person) => {
+    if (person.sex === 'm'
+      && (Math.ceil(person.died / 100) === century || !century)) {
+      return (sum * men + (person.died - person.born)) / ++men;
+    } else {
```

Comment:
If I remember right we have to avoid using 'ifs' in this task. Use ternary operator instead